### DISCUSSION TITLE
Query on exporting link_slaves_found() function

### DISCUSSION THREAD

**Discussion by vijendarmukunda** [thesofproject/sof#7882] [Q&A]
We are currently working on SoundWire enablement using SOF stack for AMD platforms.

We have used link_slaves_found() function with different arguments in our stack.
instead of "sdw_intel_ctx" , we have used our struct sdw_amd_ctx *sdw as defined below.

struct sdw_amd_ctx {
int count;
int num_slaves;
struct sdw_amd_slave_id *ids;
struct platform_device *pdev[2];
};
Please suggest us , how can we use the function with different arguments?

**Comment by vijendarmukunda** [comment]
@plbossart : could you please help to comment , how can we export "link_slaves_found" function?

**Reply by lgirdwood** [reply]
@vijendarmukunda you should be able to propose a PR with any change or extension to this API in the normal method. I dont think there is anything Intel specific about finding links.

**Reply by vijendarmukunda** [reply]
@lgirdwood : I do agree. But I was not sure about the common place holder to define this API.

**Reply by lgirdwood** [reply]
Just try and see what fits best in your patch. The code review will align everyone to best way forward.
